Mathlib/Topology/ContinuousFunction/StarOrdered.lean

```diff
@@ -0,0 +1,60 @@
+/-
+Copyright (c) 2024 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import Mathlib.Analysis.Complex.Basic
+import Mathlib.Topology.ContinuousFunction.Algebra
+
+/-! # Continuous functions as a star-ordered ring -/
+
+namespace ContinuousMap
+
+variable {α : Type*} [TopologicalSpace α]
+
+lemma starOrderedRing_of_sqrt {R : Type*} [PartialOrder R] [NonUnitalRing R] [StarRing R]
+    [StarOrderedRing R] [TopologicalSpace R] [ContinuousStar R] [TopologicalRing R]
+    (sqrt : R → R) (h_continuousOn : ContinuousOn sqrt {x : R | 0 ≤ x})
+    (h_sqrt : ∀ x, 0 ≤ x → star (sqrt x) * sqrt x = x) : StarOrderedRing C(α, R) :=
```

Comment:
Something I'm curious about; we ended up generalizing `StarOrderedRing` from "every nonnegative element is `star c * c`" to "every non-negative element is a sum of such elements".

Is there an analogy here where `sqrtPieces : R -> Finset R`, and is there any situation in which it is interesting?

Reply:
None that I know of, especially because you would need to have the same number of pieces. So it would be more like `sqrtPieces : R → ((n : ℕ) × (Fin n → R))` or something to that effect.